### PR TITLE
DPC-3691 view organization

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3691

## 🛠 Changes

- app/components/core/table/*
- app/components/core/basic*
- app/components/page/organization/show*
- app/views/organization/show using page component

## ℹ️ Context for reviewers

IP address hard coded as functionality not available.

## ✅ Acceptance Validation

Should look like attachment in jira ticket at http://localhost:3100/portal/organizations/[org-id]

Screenshot from local development.
orgview

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: dpc-portal/app/models/organization.rb [hunk 1]
```diff
@@ -2,22 +2,34 @@
...
 class Organization
-  attr_reader :api_id, :name
+  attr_reader :api_id, :name, :npi
 
   def initialize(api_id)
     @api_id = api_id
     @client = DpcClient.new
     data = @client.get_organization(api_id)
     @name = data.name
+    @npi = data.identifier.select { |id| id.system == 'http://hl7.org/fhir/sid/us-npi' }.first&.value
```
Comment (MEspositoE14s) [inline dpc-portal/app/models/organization.rb:12]: Thanks!  I was just getting ready to try and write this for my own ticket.  

How fast can we get this approved and merged? 😃

**Review by MEspositoE14s — APPROVED** [review]

File: dpc-portal/app/views/organizations/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	  <h2>Public IPs</h2>
+	</div>
+	<div class="grid-col-4 margin-top-2">
+	  <%= button_to 'Add IP', root_path, method: :get, class: 'usa-button' %>
+	</div>
+      </div>
+    </div>
+    <div class="usa-table">
+      <% if @organization.public_ips.present? %>
+        <%= render(Core::Table::Component.new(id: 'public-ips-table')) do %>
```
Comment (kyeah) [inline dpc-portal/app/views/organizations/show.html.erb:85]: Hmm this rendered table needs the `width-full` class so it fills the container. Doesn't seem like there's an obvious way to add custom classes in here though.

Could we add a new component option to either pass in custom classes, or a specific option for `full_width`?

I think there's probably a smarter way for any core component to accept custom classes using a base class, but I'm not smart enough to figure that out right now 😄 Happy to punt on fixing the table width for a future ticket.

File: dpc-portal/app/views/organizations/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            <tr>
+	      <th scope="row" role="columnheader">Label</th>
+	      <th scope="row" role="columnheader">Key ID</th>
+	      <th data-sortable scope="row" role="columnheader" aria-sort="ascending">Creation Date</th>
+	      <th scope="row" role="columnheader">&nbsp;</th>
+	    </tr>
+          </thead>
+          <%= render partial: 'public_key', collection: @organization.public_keys %>
+        <% end %>
+	<div class="usa-sr-only usa-table__announcement-region" aria-live="polite"></div>
```
Comment (kyeah) [inline dpc-portal/app/views/organizations/show.html.erb:65]: Just realized a formatting checker / autoformatter would be nice to have 🙂

File: dpc-portal/spec/support/match_html_fragment.rb [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+module MatchHtmlFragment
```
Comment (kyeah) [inline dpc-portal/spec/support/match_html_fragment.rb:3]: This is pulled from the uswds component library, right? Let's add a comment here linking to the original source 🙏

File: dpc-portal/app/views/organizations/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,104 @@
+<div>
+  <div class="shadow-5 border-1px border-base-lighter radius-md margin-bottom-4 padding-left-3 padding-bottom-3">
+    <h2><%= @organization.name %></h2>
```
Comment (kyeah) [inline dpc-portal/app/views/organizations/show.html.erb:3]: I have a feeling this should technically be an `h1` from an accessibility point of view, but we don't have accessibility checks right now s😄

**Review by kyeah — APPROVED** [review]
Nice, thank you! I added a few notes from a stylistic / accessibility point of view, but as we discussed — those shouldn't be blockers to merging and marking the ticket as done. 

We can add followup tickets for anything a little more substantial (I noted a couple nice-to-have CI/CD items for us to have as well.)

File: dpc-portal/app/views/organizations/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    <div class="grid-container">
+      <div class="grid-row">
+	<div class="grid-col-8">
+	  <h2>Public Keys</h2>
+	</div>
+	<div class="grid-col-4 margin-top-2">
+	  <%= button_to 'Create key', root_path, method: :get, class: 'usa-button' %>
+	</div>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (jdettmannnava) [reply dpc-portal/app/views/organizations/show.html.erb]: Done!

File: dpc-portal/app/views/organizations/show.html.erb (hunk 2 above)
Comment (jdettmannnava) [reply dpc-portal/app/views/organizations/show.html.erb]: Done!

File: dpc-portal/app/views/organizations/show.html.erb (hunk 3 above)
Comment (jdettmannnava) [reply dpc-portal/app/views/organizations/show.html.erb]: tabs, argh!

File: dpc-portal/app/components/core/card/basic_component.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<div>
+  <div class="shadow-5 border-1px border-base-lighter radius-md margin-bottom-4 padding-x-3 padding-bottom-3">
+    <div class="display-flex flex-row flex-justify">
+      <div class="flex-align-self-start">
+        <%= raw @text_content %>
+      </div>
+      <div class="flex-align-self-end">
```
Comment (kyeah) [inline dpc-portal/app/components/core/card/basic_component.html.erb:7]: This controls vertical alignment, so `flex-align-self-center` seems more suitable here.

Current Alignment:
CleanShot 2023-12-15 at 15 38 47@2x

Centered Alignment:
CleanShot 2023-12-15 at 15 38 37@2x

File: dpc-portal/app/components/core/card/basic_component_preview.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+module Core
+  module Card
+    # Basic Card Component
+    class BasicComponentPreview < ViewComponent::Preview
+      # button must have name, path, and method keys
+      #
+      # @param text_content textarea
+      # @param button_name textarea
+      # @param inner_html textarea
+      def parameterized(text_content: '<h1>Welcome</h1>', button_name: 'Go to thing', inner_html: '')
```
Comment (kyeah) [inline dpc-portal/app/components/core/card/basic_component_preview.rb:12]: Would recommend providing some basic inner_html content by default here!

File: dpc-portal/app/components/core/table/table_component_preview.rb [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        </tbody>
+      HTML
+
+      # `stacked` and `stacked_header` determine how the table looks on mobile-width
+      # views.
+      #
+      # @param sortable toggle
+      # @param inner_html textarea
+      # @param additional_classes textarea space-delimited
+      def parameterized(additional_classes: '', sortable: false, inner_html: DEFAULT_INNER_HTML)
```
Comment (kyeah) [inline dpc-portal/app/components/core/table/table_component_preview.rb:62]: Not sure why, but I noticed in Lookbook that enabling the `sortable` toggle doesn't do anything to the table preview 🤔

**Review by kyeah — APPROVED** [review]
nice component-ization 🌟

File: dpc-portal/app/components/core/table/table_component_preview.rb (hunk 9 above)
Comment (jdettmannnava) [reply dpc-portal/app/components/core/table/table_component_preview.rb:62]: Actually, it adds a hidden thing that is necessary for sorting to work. You can see it in the html tab. I will put that in the notes.

File: dpc-portal/app/models/organization.rb [hunk 10]
```diff
@@ -2,22 +2,34 @@
 
 # Wrapper for calls to dpc-api
 class Organization
-  attr_reader :api_id, :name
+  attr_reader :api_id, :name, :npi
 
   def initialize(api_id)
     @api_id = api_id
     @client = DpcClient.new
     data = @client.get_organization(api_id)
```
[the remark this answers is not in the document]
Comment (jdettmannnava) [reply dpc-portal/app/models/organization.rb:10]: Addressed

**Pull request merged**